src/pages/CustomerRequisitionsPage.js

```diff
@@ -7,6 +7,8 @@ import React, { useCallback } from 'react';
 import PropTypes from 'prop-types';
 import { View } from 'react-native';
 
+import { navigation } from 'react-navigation';
```

Comment:
this is undefined, I *think* we want https://github.com/react-navigation/core/blob/master/src/routers/StackRouter.js here

Reply:
Had a look through `react-navigation` and their typescript definitions. I think the best way to do it (without ts) is to just make sure the methods we use are defined on the object. 

086a2cfbe6d5bf0e7158195afd9bc3ed7e48ed7a.

Reply:
Hate to hold this up but using `shape` seems to just do nothing? It seems equivalent to using `object` except the linter doesn't complain - if i'm not wrong, why not just use `object`, disable the rule and https://github.com/openmsupply/mobile/issues/845 knock this one away with this

Reply:
> Hate to hold this up but using `shape` seems to just do nothing? It seems equivalent to using `object` except the linter doesn't complain - if i'm not wrong, why not just use `object`, disable the rule and #845 knock this one away with this

No worry about hold up. Sorry for late reply :).

As I understand it, it works like this:

```
Blah.propTypes = {
  anyObject: PropTypes.object,
  numberObject: PropTypes.objectOf(PropTypes.number), 
  fooBarObject: PropTypes.shape({ foo: PropTypes.string, bar: PropTypes.number }),
  ....
```

`Blah` takes props:

- `anyObject`, can be any object with any keys, values.
- `numberObject`, can have any keys, but all values must be numbers.
- `fooBarObject`, must have keys `foo` and `bar` and respective values of correct type.

So, AFAIK, it goes `PropTypes.object -> PropTypes.objectOf -> PropTypes.shape`, from most general to most specific. `object` doesn't do much type checking at all, so guessing that's probably why eslint doesn't like it.

Reply:
It's been a bit since I looked at this - but when I changed shape to be an arbitrary object to check it was actually doing anything, it didnt complain. It would seem obvious that specifying a shape it would check the fields, but it didnt actually seem to! Maybe i'm wrong though

Reply:
I _think_ it only complains if the fields are missing, but it doesn't prevent you adding arbitrary ones. Maybe you added fields to the object but didn't remove any?

Reply:
Ahh. I get no warnings at all with an empty object. All it does it ensure that any fields that are defined are the correct type, huh? Not great heh, sorry for not understanding

Reply:
Yeah I think so, you might be able to make them required. Finishing up prop types is top on my list after the indicator stuff, so hopefully will get around to this today/tomorrow.